### PR TITLE
Add Observation Mapping

### PR DESCRIPTION
**Pull request by sherrif10**
https://github.com/openmrs/openmrs-contrib-fhir2-ig/pull/27
Added obs mapping
obs

### CODE REVIEW

**Comment by sherrif10** [comment]
cc @pmanko  @mozzy11 . Kindly requesting for your reviews. Thanks.

File: input/fsh/omrs-observation.fsh [hunk 1]
```diff
@@ -14,6 +14,21 @@ Description: "Observation resource created by a OpenMRS that representing a resu
...
+* bodySite 0..1
+* method 0..1
+* specimen 0..1
+* device 0..1
+* referenceRange 0..*
+* hasMember 0..*
+* derivedFrom 0..*
+* component 0..*
 
 // OpenMRS Task  Example
```
Comment (pmanko) [inline input/fsh/omrs-observation.fsh:33]: The comment might be outdated since it refers to `Task`

File: input/fsh/omrs-observation.fsh [hunk 2]
```diff
@@ -31,3 +46,18 @@ Description: "Example OMRS Observation Resource"
...
+
+Mapping: ObservationMapping
+Source: OpenMRSObservation
+Target: "http://hl7.org/fhir/3.0/StructureDefinition/Observation"
+Id: Observation-mapping
+Title: "FHIR HL7 vs OMRS FHIR2"
+Description: "FHIR resource that is used to support diagnosis, monitor progress, determine baselines and patterns and even capture demographic characteristics As such, Observations have a wide range of uses and show up in a wide range of places"
+* -> "OpenMRSObservation" "FHIR Observations are mapped to OpenMRS Obs objects"
+* id -> "uuid"
+* status -> "obs.status"
```
Comment (pmanko) [inline input/fsh/omrs-observation.fsh:59]: Do we have any mappings of the value set for each of these fields? What status in openmrs maps to which FHIR status?

**Review by pmanko — APPROVED** [review]

File: input/fsh/omrs-observation.fsh [hunk 3]
```diff
@@ -14,6 +14,21 @@ Description: "Observation resource created by a OpenMRS that representing a resu
 * component 0..*
 * encounter 0..1
 * value[x]  0..1
+* partOf 0..*
```
Comment (pmanko) [inline input/fsh/omrs-observation.fsh:17]: Do we support all of these fields in the fhir2 module? Looking at the translator (https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationTranslatorImpl.java), it doesn't seem like we set `focus` or `performer` or `note`, for example.

**Review by pmanko — COMMENTED** [review]
Can you double-check that we're only adding the fields to that the fhir2 module supports?

**Review by pmanko — CHANGES_REQUESTED** [review]
Make sure the listed fields represent what the translator maps: https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationTranslatorImpl.java

File: input/fsh/omrs-observation.fsh (hunk 3 above)
Comment (sherrif10) [reply input/fsh/omrs-observation.fsh]: Hello @pmanko, We had agreed that we can include fields that even we don't  support and we can mark them with cardinality 0..0, can we go ahead and remove those we dont support.

File: input/fsh/omrs-observation.fsh (hunk 2 above)
Comment (sherrif10) [reply input/fsh/omrs-observation.fsh:55]: Yes we have, It is obs.status that maps status in openmrs .

**Comment by sherrif10** [comment]
cc @pmanko. Requesting your reviews.

**Comment by abertnamanya** [comment]
Hi @sherrif10  I did work on observation resource in detail. I think there is no need for this PR. You can close it

**Comment by sherrif10** [comment]
@abertnamanya  Where did you fix it, Could you share the merged link here?

**Comment by pmanko** [comment]
@sherrif10 @abertnamanya it seems like the two observation PRs have conflicts that need to be resolved.

**Comment by sherrif10** [comment]
Resolved conflicts. Thanks.

**Comment by sherrif10** [comment]
Hello @abertnamanya , Am closing this ticket such that you finish it in your recent ticket.

**Comment by sherrif10** [comment]
@abertnamanya , Let me know if you have time, Unless you don't, i can fix quickly.

**Pull request closed without merging**